server/src/server.ts

```diff
@@ -0,0 +1,45 @@
+import { Socket, Server } from "socket.io";
+
+export class CallForHelpServer {
+    public static readonly PORT: number = 3000;
+    private port: string | number;
+    private io: Server;
+
+    constructor() {
+        this.configurate();
+        this.createServerSocket();
+        this.listen();
+    }
+
+    private configurate(): void { 
+        const proccessPort = process.env.PORT;
+
+        if (proccessPort === undefined) {
+            this.port = CallForHelpServer.PORT;
+            console.log(`Environment variable port is undefined, defaulting to ${CallForHelpServer.PORT}`);
+        } else {
+            this.port = proccessPort;
+        }
+    }
+
+    private createServerSocket(): void {
+        this.io = require('socket.io')(this.port, {
+            cors: {
+                origin: ['http://localhost:8080']
```

Comment:
The default client url is `http://localhost:5173`, but it would also be nice to have an environment variable to that, for easier deployment.